### PR TITLE
Add ecr_repository

### PR DESCRIPTION
**Pull request by lotoussa**
| Q                 | A
| ----------------- | ---
| 🐛 Bug fix?       | no
| 🚀 New feature?   | yes
| ⚠ Deprecations?   | no
| ❌ BC Break       | no
| 🔗 Related issues | #151 
| ❓ Documentation  | no

## Description

I have already created the files adding the resource ecr_repository. I still have to do the tests, on the whole.
For the moment I have not found any need to use normaliser but do not hesitate to make remarks if necessary.

EDIT: Tests are up, but i need to execute a coverage to upgrade them.
Feel free to check the commits already pushed.

EDIT: i'm ok.

### CODE REVIEW

File: pkg/remote/aws/repository/ecr_repository.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	client ecriface.ECRAPI
+}
+
+func NewECRRepository(session *session.Session) *ecrRepository {
+	return &ecrRepository{
+		ecr.New(session),
+	}
+}
+
+func (r *ecrRepository) ListAllRepository() ([]*ecr.Repository, error) {
```
Comment (wbeuil) [inline pkg/remote/aws/repository/ecr_repository.go:23]: `ListAllRepositories`

File: pkg/remote/aws/repository/ecr_repository.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	var tables []*ecr.Repository
```
Comment (wbeuil) [inline pkg/remote/aws/repository/ecr_repository.go:24]: Name it `repositories` not tables

File: pkg/remote/aws/repository/ecr_repository_test.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+	"github.com/aws/aws-sdk-go/aws"
+
+	"github.com/stretchr/testify/mock"
+
+	"github.com/cloudskiff/driftctl/mocks"
+	"github.com/r3labs/diff/v2"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_ecrRepository_ListAllRepository(t *testing.T) {
```
Comment (wbeuil) [inline pkg/remote/aws/repository/ecr_repository_test.go:18]: `ListAllRepositories`

File: pkg/resource/aws/aws_ecr_repository_ext.go [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package aws
```
Comment (wbeuil) [inline pkg/resource/aws/aws_ecr_repository_ext.go:1]: Since there are no normalizations to do, you can remove this file, it is useless.

**Review by wbeuil — CHANGES_REQUESTED** [review]
Overall, it looks good, just small typos on naming variables and functions.

For permissions, create our limited policy and run each time driftctl with it.

File: doc/cmd/scan/supported_resources/aws.md [hunk 5]
```diff
@@ -85,6 +85,7 @@ As AWS documentation recommends, the below policy is granting only the permissio
                 "ec2:DescribeVpcClassicLinkDnsSupport",
                 "ec2:DescribeSubnets",
                 "ec2:DescribeNatGateways",
+                "ecr:DescribeRepositories",
```
[the remark this answers is not in the document]
Comment (lotoussa) [reply doc/cmd/scan/supported_resources/aws.md:88]: thank you, good point

File: pkg/resource/aws/aws_ecr_repository_ext.go (hunk 4 above)
Comment (lotoussa) [reply pkg/resource/aws/aws_ecr_repository_ext.go]: agreed

File: pkg/remote/aws/test/ecr_repository_multiple/results.golden.json [hunk 6]
```diff
@@ -0,0 +1,48 @@
+[
```
Comment (eliecharra) [inline pkg/remote/aws/test/ecr_repository_multiple/results.golden.json:1]: It's a good practice to add within this directory the terraform file you used to generate theses resources.

If someone has to maintain your test in the future for example.

File: pkg/resource/aws/aws_ecr_repository_test.go [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+					if err != nil {
+						t.Fatal(err)
+					}
+				},
+				Check: func(result *acceptance.ScanResult, stdout string, err error) {
+					if err != nil {
+						t.Fatal(err)
+					}
+					result.AssertDriftCountTotal(1)
+					result.AssertDeletedCount(0)
```
Comment (eliecharra) [inline pkg/resource/aws/aws_ecr_repository_test.go:56]: You can also ensure they are no unmanaged resources there

**Review by eliecharra — CHANGES_REQUESTED** [review]
Awesome work 👍🏻 Some small minor remarks

File: pkg/remote/aws/test/ecr_repository_multiple/results.golden.json (hunk 6 above)
Comment (lotoussa) [reply pkg/remote/aws/test/ecr_repository_multiple/results.golden.json:1]: right

**Review by eliecharra — APPROVED** [review]

**Review by wbeuil — APPROVED** [review]

**Pull request merged**